Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -2045,6 +2046,9 @@ else if (!dasSnapshotFile.canRead()) {
       catch (MissingResourceException e) {
         // NOP
       }
+      if (DartConfigurable.getMLCodeCompletionEnabled(myProject)) {
```

Comment:
I'm wondering how this will interact with the analysis server default for ML completion, if we at some point choose to change it. Should we pass in `--no-enable-completion-model` if the box is unchecked? Or, how would user indicate to disable ML completion if the analysis server later enables it by default?

Reply:
One idea to address this issue is to never enable it by default in analysis server and to instead enable it by default across clients. But I'm not sure if that's the best rollout strategy.

Reply:
That's feasible, and avoid timing issues with rolling it out between the analysis server and clients. We may consider some other way to enable these flags in the future - to have a set of settings in the analysis server that we can query and modify.